### ISSUE TITLE
Add "rack" to type of bicycle parking quest

### ISSUE THREAD

**Issue by pkoby** [westnordost/StreetComplete#1038]
I did some surveying lately, and was annoyed with the "What is the type of this bicycle parking" quest. The choices given seem a little strange. I understand the choice of Stands and Wheelbenders (which I assume tags as =wall_loops), but then Shed, Lockers, and Building are weird choices with few use cases for me. 

By far the majority of bicycle parking around here would be tagged as =rack. This type would include the coathanger racks (https://commons.wikimedia.org/wiki/File:Triton_Bike_Rack.png), comb-style racks, and probably the wavy style (https://commons.wikimedia.org/wiki/File:Serpentine_Bike_Rack_Powder-Coat.png).

Looking at Taginfo (https://taginfo.openstreetmap.org/keys/bicycle_parking#values), stands and wall_loops are clearly good choices, but shed, lockers, and building don't equal the same percentage of usage as rack, even put together.

Could Rack be added as the third option?

**Comment by matkoniecz** [comment]
> Wheelbenders (which I assume tags as =wall_loops)

You are right.

**Comment by matkoniecz** [comment]
> comb-style racks

I have no idea what is this.

> and probably the wavy style

Depending is size it is either exact match for to wall_loop (one is unable to attach bicycle frame to it) or exact match to stand (one is able to attach bicycle frame to it, track supports bicycle).

> coathanger racks 

I am less sure here, as this type is not appearing where I live - but I would classify it as a weird stand (it is possible to attach bicycle frame to it that is supported by rack structure, right?)

**Comment by westnordost** [comment]
Basically, answer "stands" if you can attach your bike frame to it, answer "wheelbenders" if you attach your wheel to it. That is the distinction that matters.

Do you think that the quest should be clearer on this? I.e. I thought that people would understand "stands" as an umbrella term.

**Comment by pkoby** [comment]
Interesting. I have always referred to https://wiki.openstreetmap.org/wiki/Key:bicycle_parking for my tagging, and the notable thing about "stands" is that it states on the wiki that you can lock your frame and a wheel to it. With many racks (aside perhaps from the wavy style), it's not easy to lean the frame against it, but you're sort of inserting the front of the bike into the rack. You can lock the frame to it, but it's mostly held by the wheel or headtube. Considering the coathanger rack style, it's possible to lean the frame on the outside bars, but not on the inside, but I wouldn't consider the inside spots wheelbenders either.

In other words, it's a vague sort of thing. This section of the talk page does think there should be a distinction: https://wiki.openstreetmap.org/wiki/Talk:Key:bicycle_parking#bicycle_parking.3Drack.3F

I would say clarification would help, because it wasn't obvious to me at least, but even so, I'm not sure I'd use "stands" for most of the parking around here. I think the major distinction is multi-bike (like more than two capacity) parking that's not using just the wheel.

I guess you'd also consider "bollard" as a type of stand, despite the distinction in the wiki? I have no argument there, at least.

**Comment by matkoniecz** [comment]
> multi-bike (like more than two capacity) parking

I think that it does not matter whatever individual stands are made from separate metal parts or is it manufactured add one part.

Especially as individual stands may be connected by metal part that is underground - why it should change anything?

Adding "can't say" section titled "bicycle parking is not matching any photo" may be a good idea.

**Comment by westnordost** [comment]
I was thinking of adding a few words on these two entries:
- wheelbenders (lockable only to wheel)
- stands (lockable to frame)

Would this wording be clear to you? (I am not a native speaker)

**Comment by gappleto97** [comment]
Which would this fall under?

![](https://framapic.org/BQARCXW1gKw1/JYLDuj5Kg6eJ)

Because this is the type I see most often, and its not obviously covered by any of the pictures.

**Comment by pkoby** [comment]
That's what I'd call a rack, and it's what I mentioned earlier as "comb" style. Around where I live, it's mostly coathangers, these, and wavy types. Almost no "staple" stands. Some bollard type stands, though.

**Comment by matkoniecz** [comment]
wheelbender (AKA =wall_loop), as structure is not supporting bicycle frame and it is not possible to attach bicycle frame to the parking (except in cases of placing it in parallel to the bicycle stand).

**Comment by westnordost** [comment]
See @matkoniecz 

Any comments on my suggestion?

**Comment by westnordost** [comment]
Absolutely. One of the reasons why there are no separate options. See also https://github.com/westnordost/StreetComplete/pull/923#issuecomment-370848376

**Comment by pkoby** [comment]
Thanks for the discussion. I suppose that takes care of this issue.

**Comment by westnordost** [comment]
I did not change anything yet!

**Comment by pkoby** [comment]
Whoops! Sorry.

**Comment by westnordost** [comment]
So, would look like this then:

![device-2018-05-01-183223](https://user-images.githubusercontent.com/4661658/39482106-06ca3f40-4d6e-11e8-89f6-c7565f02d749.png)

**Comment by gappleto97** [comment]
Really, I would just settle for having additional pictures, even if they just map back to "stand". Visuals are going to be far more explanatory than words.

Maybe borrow the road surface UI for this?

**Comment by westnordost** [comment]
but these would be a whole lot of pictures then

**Comment by goldfndr** [comment]
To me, a "stand" is limited to 2 bicycles, and a "rack" can handle 4 or more; a rack has at least three vertical bits (when viewed relative to the direction bicycles are to be parked in) while a stand has 1 or 2 (depending on angle) (and bollard has 1).

Regarding @gappleto97's "comb", I'd tag that as "wall_loops" if a tire can't fit through a slot and "rack" if a tire does fit through.

I spot-checked a few "anchors" and they were actually "stands". Doesn't mean that "anchors" don't exist, obviously. 

@matkoniecz mentioned that individual stands may be connected by a metal part that is underground. Is there a retail example? I'd think that extra metal that needs to be buried and enlarges the package wouldn't be preferable.

**Comment by rugk** [comment]
Just as a side note, I would also leave out the "=" in the brackets. That is just redundant.

**Comment by pkoby** [comment]
Looks good. After this discussion, I kind of think that the whole bicycle_parking=* should be amended, but that's a whole other issue. I think making almost all cases =stands or =wall_loops is not a bad idea. I don't like how =wall_loops lumps together a lot of things that have nothing to do with a wall, though. Perhaps =wheelbenders would be best, but that would deprecate a lot of points....

**Comment by matkoniecz** [comment]
> I don't like how =wall_loops lumps together a lot of things that have nothing to do with a wall, though. Perhaps =wheelbenders would be best, but that would deprecate a lot of points....

It is the same problem as highway=unclassified - it is probably too late to change it.

**Comment by westnordost** [comment]
What's wrong with `highway=unclassified`?

**Comment by westnordost** [comment]
I always assumed that it meant that it is a road that is not officially classified.

**Comment by matkoniecz** [comment]
From quick look ( http://overpass-turbo.eu/s/yyy ) it appears that there are highway=unclassified with ref tag in UK, but I am not sure is it a tagging mistake or examples that this rule is not true.

From looking at https://wiki.openstreetmap.org/wiki/Tag:highway%3Dunclassified it seems that there is no reason to assume that highway=unclassified has no ref tag.

Also, there is no good reason to change highway tag value for road just because there is a number sign posted next to it.

**Comment by pkoby** [comment]
I agree that highway=unclassified sounds like it should be put into some category later when further info is known. But around where I live, there isn't any official classification for really any roads. Highway=residential is not designated as such by any governmental authority, it's simply a class based on what the area around is like, speed limit, and such. For my area, it seems like highway=unclassified is very similar to highway=residential, but used when there isn't a residential area nearby.

**Comment by HolgerJeromin** [comment]
"unclassified" is the official name of this type of road in the uk. The first mappers lived there so it seemed natural to use uk road class names.
But too late to change :-)
